include/onnxruntime/core/common/logging/isink.h

```diff
@@ -13,6 +13,11 @@ class ISink {
  public:
   ISink() = default;
 
+  enum SinkType { BaseSink,
+                  CompositeSink,
+                  EtwSink };
+  virtual SinkType GetType() const { return BaseSink; }
```

Comment:
nit: could be const property of base class that is set in ctor instead of a virtual method. assuming it won't change in the lifetime of a Sink.